crates/iota-sdk/src/apis/governance.rs

```diff
@@ -58,12 +58,21 @@ impl GovernanceApi {
     /// the protocol version, the reference gas price, the total stake, active
     /// validators, and much more.
     pub async fn get_latest_iota_system_state(&self) -> IotaRpcResult<IotaSystemStateSummary> {
-        Ok(self
-            .api
-            .http
-            .get_latest_iota_system_state()
-            .await
-            .map(Into::into)?)
+        let mut version = semver::Version::parse(&self.api.info.version)
+            .unwrap_or(NODE_SOFTWARE_VERSION_PROTOCOL_VERSION_5);
+        // Ignore any pre-release version as `get_latest_iota_system_state_v2` is
+        // available for `0.11.0-alpha` as well.
+        version.pre = semver::Prerelease::EMPTY;
+        if version < crate::NODE_SOFTWARE_VERSION_PROTOCOL_VERSION_5 {
```

Comment:
I would do this differently. Instead of relying on the version we should look into the rpc methods exposed to check if the v2 is available on the server.
```
struct ServerInfo {
    rpc_methods: Vec<String>,
    subscriptions: Vec<String>,
    version: String,
}
```

Reply:
Hardcoding the node-software version is indeed a a downside of the current patch. I tried to keep changes to a minimum considering the following implications:

* The name of the method follows the `openprc` overrides in `GovernanceReadApi` . This has to be hardcoded eitherway.
* Looking up the method requires searching a sequence of methods, which is now represented through a `Vec`. We could use `BTreeSet` but this has implications to the public API.

Will delegate resolving this comment to @Thoralf-M .

Reply:
@alexsporn why do you think that checking the version is not sufficient?

Reply:
It will get messy very quickly and suddenly we are going to start adding version constants and more and more version checks which will make the code messier instead of following the pattern.

```
if server.supports(rpc_v2) {
    rpc_v2() 
} else {
    rpc_v1()
}
```

Which is a valid check independent of any version numbers and schemes.

Version checks like the proposed one also put a condition on the indexer or any other custom RPC extension to follow the exact same versioning as the node software, because we rely on the node version to check for RPC compatibility instead of e.g. the protocol version that introduced the required change.